File: src/main/java/com/dv/telegram/command/ListCommands.java
package com.dv.telegram.command;

import com.dv.telegram.WikiBot;
import org.apache.commons.lang3.StringUtils;

import java.util.ArrayList;
import java.util.List;

public class ListCommands extends BasicBotCommand {

    @Override
    public String getName() {
        return ListCommands.class.getSimpleName();
    }

    @Override
    public String getDescription(WikiBot bot) {
        return String.format(
            "`%s %s` — получить список команд бота.",
            bot.getBotName(),
            getCommandText()
        );
    }

    @Override
    public boolean useMarkdownInResponse() {
        return true;
    }

    @Override
    public String getDefaultCommandName() {
        return "/listCommands";
    }

    @Override
    public String getResponse(String text, WikiBot bot) {
        List<String> lines = new ArrayList<>();

        lines.add("Список команд бота:");

        for (BotCommand command : bot.getSpecialCommands().getCommands()) {
            lines.add(String.format(
                "— `%s %s`",
                bot.getBotName(),
                command.getCommandText()
            ));
        }

        lines.add(String.format(
            "Для получения справки по команде используйте команду%n`%s %s <commandName>`",
            bot.getBotName(),
            bot.getSpecialCommands().getHelpCommand().getCommandText()
        ));

        return StringUtils.join(lines, "\n\n");
    }
}

File: src/main/kotlin/com/dv/telegram/command/ListCommands.kt
package com.dv.telegram.command

import com.dv.telegram.WikiBot
import org.telegram.telegrambots.meta.api.objects.Update

class ListCommands : BasicBotCommand() {
    override val name: String = javaClass.simpleName

    override fun getDescription(bot: WikiBot) =
        "`${bot.botName} $commandText` — получить список команд бота."

    override fun useMarkdownInResponse() = true

    override val defaultCommandName = "/listCommands"

    override fun getResponse(text: String, bot: WikiBot, update: Update): String {
        val lines = mutableListOf<String>()

        lines.add("Список команд бота:")

        for (command in bot.specialCommands.commands) {
            lines.add(
                "— `${bot.botName} ${command.commandText}`"
            )
        }

        lines.add(
            "Для получения справки по команде используйте команду\n`${bot.botName} ${bot.specialCommands.helpCommand.commandText} <commandName>`"
        )

        return lines.joinToString("\n\n")
    }
}
